main.js

```diff
@@ -1,75 +1,72 @@
 const fetch = require("node-fetch");
 
-const cachedTemprs = { device: {}, schedule: {} };
+const cachedTemprs = {};
 const currentRequests = {};
 
-const makeRequest = (url, config) => {
+function makeRequest(url, config) {
     if (!(url in currentRequests)) {
         currentRequests[url] = fetch(url, {
             headers: { "X-Core-Token": config.oopCoreToken }
-        })
-            .then(async res => {
-                setTimeout(() => {
-                    delete currentRequests[url];
-                }, 0);
+        }).then(async res => {
+            setTimeout(function() {
+                delete currentRequests[url];
+            }, 0);
 
-                const json = await res.json();
+            const json = await res.json();
 
-                return json;
-            });
+            return json;
+        });
     }
 
     return currentRequests[url];
-};
+}
 
 module.exports = (broker, config, logger) => {
-    broker.consume(config.temprInputQ, message => {
+    function queueTemprs(temprs, data) {

```

Comment:
I would move this outside of the export function and add `config` as an argument here.

Reply:
Also, functions should not be declared like this within a scope, as it leads to unpredictable behavior. If you don't want to use an arrow function you can say `const queueTemprs = function(...) { ....`